third_party/nvfuser/csrc/executor_kernel_arg.h

```diff
@@ -284,6 +284,35 @@ class TORCH_CUDA_CU_API KernelArgumentHolder {
     return index_mode_;
   }
 
+  // Allows the user to specify a larger index mode than reqd
+  void setIndexMode(KernelIndexMode mode) {
+    // index_mode_ is deduced on args. We won't allow setting it to a smaller
+    // size than required
+    auto smallest = getSmallestIndexModeRequired();
+
+    // well, turns out index type was too small, regardless of the argument.
+    if (smallest == KernelIndexMode::INT64 &&
+        index_mode_ != KernelIndexMode::INT64) {
```

Comment:
Yes, we don't hit that. It would happen if someone changes `index_mode_` without re-doing the argument marshaling for tensors. That's exactly an error that I happen to make - as I didn't knew that KernelArgumentHolder does it when you `push` tensors

Reply:
that's an easy recovery to prevent kernel crash - but I can remove it if you want

Reply:
I wonder which would be better between the recovery like you does in this PR or just throwing an error. Unless there's a legitimate situation we can't avoid the inconsistent state and we need to recover from it, an outright error seems better to me. So, the question is, how exactly did you manage to hit the error?

Reply:
You're right, I updated the code to throw an internal error.

Reply:
The only way to get in the invalid state would be to push a 64b tensor into the args before setting changing the index type, in `KernelArgumentHolder::push()`.